### ISSUE TITLE
Change in package/exclude usage?

### ISSUE THREAD

**Issue by sbstjn** [serverless/serverless#3869]
Have there been changes in how to exclude paths from being included in the package? I see different archive sizes (*22kb* compared to *6.7mb*) when using `1.15.3` and `1.16.1`:

## Config

```yaml
package:
  exclude:
    - node_modules/**
```

Based on the [docs](https://serverless.com/framework/docs/providers/google/guide/packaging/#exclude--include) it should work like this.

## Using 1.15.3

Archive is ~22kb.

```bash
$ > yarn add serverless@1.15.3
$ > $(npm bin)/sls version && \
    $(npm bin)/sls package && \
    ls -ahl .serverless/sls-example.zip

1.15.3

Serverless: Packaging service...

-rw-r--r--  1 sbstjn  staff    22K Jun 27 21:58 .serverless/sls-example.zip
```

## Using 1.16.1

Archive is ~7mb

```bash
$ > yarn add serverless@1.16.1
$ > $(npm bin)/sls version && \
    $(npm bin)/sls package && \
    ls -ahl .serverless/sls-example.zip

1.16.1

Serverless: Packaging service...

-rw-r--r--  1 sbstjn  staff   6.7M Jun 27 22:00 .serverless/sls-example.zip
```

## NPM Dependencies

After moving the `serverless` package from `dependencies` to `devDependencies` in my `package.json`, the archive has `22kb` using `1.16.1` as well.

### 22kb

```json
{
  "devDependencies": {
    "serverless": "1.16.1"
  }
}
```

### 7mb

```json
{
  "dependencies": {
    "serverless": "1.16.1"
  }
}
```

### Expectation

I expected it does not matter if packages are in `dependencies` or `devDependecies` when using `package/exclude` in my configuration to exclude the `node_modules` folder.

**Comment by arabold** [comment]
Something is definitely off for us as well. I'm still trying to figure out what exactly it is, but in combination with `serverless-webpack` we're suddenly missing libraries in the `node_modules` folder. Some are present, others aren't, but all are listed in the `dependencies` explicitly.

**Comment by pmuens** [comment]
Nice catch @sbstjn 👍!

I'm pretty sure that this is related to https://github.com/serverless/serverless/pull/3737.

Could be that the exclusion patterns for the Node.js dev dependencies overwrite the ones which are set by `package.exclude`. 🤔

**Comment by pmuens** [comment]
Thanks again for reporting @sbstjn 👍 

We've just started a PR which will include a fix for this (will be updated frequently). It can be found here: https://github.com/serverless/serverless/pull/3889

**Comment by ajkerr** [comment]
After upgrading to 1.16.1 from 1.14.0, I also noticed this ballooning in size of our zip files. It turns out that *every* node_modules directory in my service is being included in the zip file for every single function. 

Our global package settings are set like so:

`
package:
  individually: true
  exclude:
    - '**/*'
`

Each function then re-includes its code.

With 1.16.1, all node_modules directories from every function in my service is being included in every zip file, despite the exclude filter above, and the fact that we are trying to package functions individually.

I tried setting excludeDevDependencies to false, but this didn't appear to have any effect.

**Comment by pmuens** [comment]
Thanks for commenting @ajkerr 👍 

We're currently working on a bug fix in #3889. We'll do a patch release once fixed.

The `excludeDevDependencies` config will be added in `v1.17` which we're about to release today. That should give you a way to opt out of this until it's fixed.
